src/plugins/intel_gpu/src/graph/graph_optimizer/prepare_buffer_fusing.cpp

```diff
@@ -602,5 +603,36 @@ void prepare_buffer_fusing::run(program& p) {
 
             node.can_be_optimized(can_reshape_be_optimized(node));
         });
+        program_helpers::do_for_types<read_value>(*node, [](read_value_node& node) {
```

Comment:
This optimization for `read_value` seems to be skipped by can_optimize check (`if(!can_optimize(node)) continue;`) above because `read_value` is dynamic at build time in 1st inference.

Reply:
Added a special case for `read_value` primitive in `can_optimize` check